code/src/cljs/sixsq/nuvla/ui/routing/events.cljs

```diff
@@ -1,16 +1,66 @@
 (ns sixsq.nuvla.ui.routing.events
   (:require [re-frame.core :refer [reg-event-fx]]
+            [reitit.frontend.controllers :as rfc]
             [sixsq.nuvla.ui.main.spec :as main-spec]
-            [sixsq.nuvla.ui.routing.utils :refer [add-base-path]]
+            [sixsq.nuvla.ui.routing.effects :as fx]
+            [sixsq.nuvla.ui.routing.utils :as utils]
             [taoensso.timbre :as log]))
 
+(reg-event-fx
+  ::navigate-back
+  (fn []
+    {::fx/navigate-back! []}))
+
+(reg-event-fx
+  ::push-state-by-path
+  (fn [_ [_ new-path]]
+    {::fx/push-state new-path}))
+
+(reg-event-fx
+  ::navigated
+  (fn [{db :db} [_ {:keys [path query-params] :as new-match}]]
+    (let [old-match                  (:current-route db)
+          controllers                (rfc/apply-controllers (:controllers old-match) new-match)
+          new-match-with-controllers (assoc new-match :controllers controllers)]
+      {:db                   (-> db (assoc :current-route new-match-with-controllers
+                                           ::main-spec/nav-path (utils/split-path-alias path)
+                                           ::main-spec/nav-query-params query-params))
+       :fx                   [[:dispatch [:sixsq.nuvla.ui.main.events/bulk-actions-interval-after-navigation]]]
```

Comment:
Yes, that one was the reason I rejected my refactoring :D I did it too, but somehow using the full namespace here wasn't enought.